File: Leetcode/153-Find-Minimum-in-Rotated-Sorted-Array/FindMinimumInRotatedSortedArray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    var left = 0
    var right = nums.length-1
    var target = nums[right]
    var mid
    while (left+1<right) {
        mid = left + Math.floor((right-left)/2)
        if (nums[mid]<= target) {
            right = mid
        } else {
            left = mid
        }
    }
    if (nums[left] <= target) {
        return nums[left]
    }
    return nums[right]
};

File: Leetcode/153-Find-Minimum-in-Rotated-Sorted-Array/FindMinimumInRotatedSortedArray.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0;
        int right = nums.size()-1;
        int target = nums[right];
        while (left + 1 < right) {
            int mid = left + (right-left)/2;
            if (nums[mid] <= target) {
                right = mid;
            } else {
                left = mid;
            }
        }
        if (nums[left] <= target) {
            return nums[left];
        }
        return nums[right];
    }
};
